actix/src/sec-5-arbiter.md

```diff
@@ -15,89 +15,148 @@ for your actors to run inside. When you call `start()` on your actor it is then
 running inside of the System Arbiter's thread. In many cases, this is all you
 will need for a program using Actix.
 
+While it only uses one thread, it uses the very efficient event loop pattern
+which works well for asynchronous events. To handle synchronous tasks that
+require heavy CPU power, it's better to avoid blocking the event loop and
+instead offload the computation to other threads. For this usecase, read the
+next section and consider using [`SyncArbiter`](./sec-6-sync-arbiter.md).
+
+## The event loop
+
+> If you're familiar with event loops in other paradigms, such as libuv (upon
+> which Node runs) or tokio (upon which Actix is implemented), each Arbiter has
+> a similar processing paradigm.
+
+> Specifically, if you're familiar with how JavaScript handles asynchronicity
+> and how the single-threaded model makes it so JS mostly doesn't need
+> synchronization primitives, this should feel very familiar to you.
+
+One `Arbiter` is in control of one thread with one task queue, and when tasks
+get spawned in an Actor's context or using `Arbiter::spawn`, the task gets
+queued up for execution in that thread.
+
+If you pay close attention to way both `ActorFuture` and the methods in the
+`Actor` trait specify their parameters, actor behavior always references both
+the actor and the context as mutable: `&mut self` (or `&mut A` in `ActorFuture`)
+and `&mut Self::Context`. This is because all of these tasks have **exclusive**
+rights to the actor and its context. This is achieved by putting all actor logic
+in the same thread, so we know no two pieces of logic involving the actor are
+running concurrently without having to use synchronization primitives.
+
+While the library is written with an asynchronous, event-driven flow, and while
+the library otherwise enables concurrency, each Actor specifically runs on one
+thread. Each task gets coordinated by the Arbiter to run eventually, but always
+on the same thread. So when a Future is spawned on the Arbiter (using
+`Arbiter::spawn` or the more common `Context::spawn`), it will execute within
+the same execution context (thread), allowing the Future full access to the
+state while avoiding synchronization (such as `Mutex`es).
+
+When you create a new Arbiter, this creates a new execution context for Actors.
```

Comment:
You can't create a new arbiter, you only can create one via `System::run`, so you'll only ever have the system Arbiter *or* SyncArbiters available to you.

Reply:
I don't think you're right, at least according to the documentation: https://docs.rs/actix/0.7.10/actix/struct.Arbiter.html#method.start

Reply:
I was going to write a section in this chapter on making new Arbiters, and was told that it's not possible to make new ones. I also looked at this process for the future 0.8.x version with similar results. So this part of the chapter was scrapped. That's why I'm saying there will only be one.

I would love to be wrong about this, so a section can be added to this chapter about how to make multiple Arbiters and allow cross communication though ...

Reply:
I am almost positive at this point I am correct about this.

[`Arbiter::start`](https://docs.rs/actix/0.7.10/src/actix/arbiter.rs.html#215-221) calls [`ArbiterBuilder::start`](https://docs.rs/actix/0.7.10/src/actix/arbiter.rs.html#337) goes to [`Arbiter::start_with_builder`](https://docs.rs/actix/0.7.10/src/actix/arbiter.rs.html#231) which calls [`Arbiter::new_with_builder`](https://docs.rs/actix/0.7.10/src/actix/arbiter.rs.html#231) which spawns a new thread and runs the tokio runtime on it. That said, maybe this doesn't work? I haven't tested it. I'll try to test it tomorrow if I get the time.

Reply:
@Firstyear I think we had misunderstanding. By not being able to make own `Arbiter` I meant you cannot make own type of `Arbiter`, not inability to create new instance of `Arbiter` 😄

Reply:
Hahah okay. In that case, instead of adding this here, can we make an extra section about the creation of new Arbiter instances with examples associated?

Reply:
Depending on what you want to say about it. There is per se not much to say about it

Reply:
What I want to say is "Here is how you make an extra Arbiter, and here is how you assign the new Actor to run on that Arbiter ..." with example code? I think there is plenty to say about it, especially the how you determine which actor runs on which Arbiter ... today I certainly don't know how to achieve that :)

Reply:
Well I guess it would be fine to add, but Actor doesn't need to know on which arbiter it runs.

Reply:
Well, I need to know how to say "Actor A run on Arbiter A, and Actor B run on Arbiter B" is more my point here ....

Reply:
@DoumanAsh But `Actor`s are run on specific arbiters, right? So the human writing the program can know which arbiter it's run on.

Reply:
@rivertam Exactly - so I think we make a later PR with an example that demonstrates this correctly.